.github/workflows/open-pr.yml

```diff
@@ -75,11 +75,22 @@ jobs:
             core.setSecret(accessToken)
             core.setOutput('token', accessToken)
       - name: initialize bare SDK clone
+        id: clone-g4w-sdk
         shell: bash
         run: |
           git clone --bare --depth=1 --single-branch --branch=main --filter=blob:none \
-            https://github.com/git-for-windows/git-sdk-64 .tmp
+            https://github.com/git-for-windows/git-sdk-64 .tmp &&
+          echo "rev=$(git -C .tmp rev-parse HEAD)" >>$GITHUB_OUTPUT
+      - name: Cache git-sdk-64 subset
+        id: cache-g4w-sdk
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-g4w-sdk
+        with:
+          path: .sdk
```

Comment:
When does the cache action save the cache?

Reply:
It's the [post checkout](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4079002007/jobs/7029893303#step:25:1) step that is implied by `actions/cache`.

Reply:
That's not ideal. It's caching the `.sdk` directory after we already cloned the target repo, which could cause collisions when opening PRs for multiple packages in the same repo. Maybe we should use `actions/cache/restore` and `actions/cache/save` to explicitely create the cache after the sparse checkout?

Reply:
Valid point.

I changed it. To prove that it works, I ran a dry-run that [did not find a cached artifact](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4087945043/jobs/7049054932#step:5:15) and [successfully cached it](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4087945043/jobs/7049054932#step:7:18), followed by a dry-run that [found the cache](https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4087967025/jobs/7049101774#step:5:15) and successfully avoided re-checking out and re-caching it.